src/GitHub.App/ViewModels/GistCreationViewModel.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using GitHub.Exports;
+using System.ComponentModel.Composition;
+using System.Reactive.Linq;
+using GitHub.Api;
+using GitHub.Factories;
+using GitHub.Primitives;
+using GitHub.Services;
+using Octokit;
+using ReactiveUI;
+using NullGuard;
+
+namespace GitHub.ViewModels
+{
+    [ExportViewModel(ViewType=UIViewType.Gist)]
+    [PartCreationPolicy(CreationPolicy.NonShared)]
+    public class GistCreationViewModel : BaseViewModel, IGistCreationViewModel
+    {
+        readonly ISelectedTextProvider selectedTextProvider;
+        readonly IApiClient apiClient;
+
+        [ImportingConstructor]
+        public GistCreationViewModel(ISelectedTextProvider selectedTextProvider, IApiClientFactory apiClientFactory)
+        {
+            Title = Resources.CreateGistTitle;
+            this.selectedTextProvider = selectedTextProvider;
+            this.apiClient = apiClientFactory.Create(HostAddress.GitHubDotComHostAddress);
```

Comment:
We actually want to do the gist on whatever server the user is connected to, which might be an enterprise instance. You can grab the correct `IApiClient` by copying what the other viewmodels are doing, like https://github.com/georgebearden/VisualStudio/blob/gist-support/src/GitHub.App/ViewModels/RepositoryCloneViewModel.cs#L39

The (private) importing constructor there takes a `IConnectionRepositoryHostMap` and calls the public constructor with the host that matches the connection that's been set by the UIController in the UIProvider service list (either the connection that was passed to RunUI, or the first logged in connection). Just having the pattern of those two constructors will give you the `IRepositoryHost`, which exposes a `ApiClient` property for that particular host.

Something else to be considered is that we likely want to create the gist on the server that corresponds to the currently opened repository, instead of letting it pick whichever one is first (ideally, we'd have an account dropdown to pick where it should create it, but that's a nice extra for later methinks)
To get the correct `IConnection`, in the command handler, you can get an instance of `ITeamExplorerServiceHolder` and get the `ActiveRepo` from it, and then grab the connection from the `ConnectionManager` connections list and pass it on to `RunUI`, more or less like https://github.com/georgebearden/VisualStudio/blob/gist-support/src/GitHub.VisualStudio/Base/EnsureLoggedInSection.cs#L53 does when it calls `IsLoggedIn`. That goes to https://github.com/georgebearden/VisualStudio/blob/gist-support/src/GitHub.Exports.Reactive/Extensions/ConnectionManagerExtensions.cs.

We probably also want to show the server name in the view, so people know where it's going to be created on.